### ISSUE TITLE
1.1.1 segfault: m_verifying bitfield can have incorrect length

### ISSUE THREAD

**Issue by wlritchi** [arvidn/libtorrent#1623]
libtorrent version (or branch): 1.1.1

platform/architecture: Arch Linux x86_64

compiler and compiler version: gcc 6.3.1 20170109

On some code paths, the `m_verified` bitfield is updated to the correct length, but the `m_verifying` bitfield is not. This can lead to a segfault when attempting to access the `m_verifying` bitfield.

See the update to `m_verified` here: https://github.com/arvidn/libtorrent/blob/libtorrent-1_1_1/src/torrent.cpp#L7157

I would expect to see a similar line below it resizing `m_verifying`.

**Comment by arvidn** [comment]
what kind of segfaults do you get? I mean, where?

**Comment by arvidn** [comment]
I rolled your suggested fix into this patch. would you mind confirming that it solves it for you?
https://github.com/arvidn/libtorrent/pull/1624

**Comment by wlritchi** [comment]
I had locally applied a patch doing the same thing as you've now rolled into #1624 and that seems to have resolved the issue for me. I can't be 100% certain because the segfaults were sporadic, sometimes hitting as soon as I start qBittorrent, sometimes going hours without issue. My layman's guess would be that some subset of my torrents were affected, and the segfault only occurred when there was some activity on one of those torrents

I ran my patched version for several days without experiencing another segfault, which is longer than it had successfully run before at any point. This makes me believe that the patch (and by extension, your patch above) solves the problem.

However, I tried reverting back to an unpatched copy of libtorrent so I could capture a full stack trace of the segfault for your request (my fault for not saving a copy the first time I went bughunting), and I haven't experienced one over the last few days. My best guess is that somehow my .fastresume files were malformed/legacy/etc in a way that could trigger a segfault without this patch, but with the patch in place, libtorrent was alive long enough to re-save them and the new copies don't hit this issue. I've got a backup copy of the .fastresume files from before running the patched copy; I'll compare them against my current ones to see if anything stands out that could be used as a test case in the future.

TL;DR confirming that the fix works, will investigate the fastresume files a bit. If I get a segfault I'll attach the stack trace here for posterity.
